notifications-sample/src/extension.ts

```diff
@@ -0,0 +1,63 @@
+import * as vscode from 'vscode';
+
+export function activate(context: vscode.ExtensionContext) {
+
+	// Simple notifications
+	let showInfoNotification = vscode.commands.registerCommand('notifications-sample.showInfo', () => {
+		vscode.window.showInformationMessage('Info Notification');
+	});
+
+	let showInfoNotificationAsModal = vscode.commands.registerCommand('notifications-sample.showInfoAsModal', () => {
+		vscode.window.showInformationMessage('Info Notification As Modal', { modal: true });
+	});
+
+	let showWarningNotification = vscode.commands.registerCommand('notifications-sample.showWarning', () => {
+		vscode.window.showWarningMessage('Warning Notification');
+	});
+
+	let showErrorNotification = vscode.commands.registerCommand('notifications-sample.showError', () => {
+		vscode.window.showErrorMessage('Error Notification');
+	});
+
+	// Notifcation with actions
+	let showWarningNotificationWithActions = vscode.commands.registerCommand('notifications-sample.showWarningWithActions', () => {
+		vscode.window.showWarningMessage('Warning Notification With Actions', 'Action 1', 'Action 2', 'Action 3').then(selection => { vscode.window.showInformationMessage(`You selected: ${selection}`, { modal: true }); });
```

Comment:
You could make this async and then use await. This line is really long so hard to see what's happening. I like async await because then 
```
const selection = await vscode.window...;
vscode.showInformationMessage(`You selected: ${selection}`);
```
is more comprehendible